src/plugins/discover/public/application/main/discover_main_app.tsx

```diff
@@ -96,21 +97,23 @@ export function DiscoverMainApp(props: DiscoverMainProps) {
   }, [resetSavedSearch, savedSearch]);
 
   return (
-    <DiscoverLayoutMemoized
```

Comment:
LGTM, only thing to consider, adding the `EuiErrorBoundary` in `discover_router.tsx` would also cover executions in routing, context and doc view. So it would be one `EuiErrorBoundray` to catch em all 🧙

Reply:
Would that not be too aggressive?

Reply:
Would there be a disadvantage to catch more potential failures causing a empty Discover rendering? Seen other plugins doing it too:
https://github.com/elastic/kibana/blob/64f30a224e52e58e8a806bba9194123ecc109860/x-pack/plugins/osquery/public/application.tsx#L63

Reply:
change LGTM, will test and then approve